app/lib/efile/nj/nj1040_calculator.rb

```diff
@@ -326,19 +326,19 @@ def calculate_line_39
       end
 
       def is_ineligible_or_unsupported_for_property_tax_credit
-        StateFile::NjHomeownerEligibilityHelper.determine_eligibility(@intake) != StateFile::NjHomeownerEligibilityHelper::ADVANCE ||
+        StateFile::NjHomeownerEligibilityHelper.determine_eligibility(@intake) == StateFile::NjHomeownerEligibilityHelper::INELIGIBLE ||
           Efile::Nj::NjPropertyTaxEligibility.ineligible?(@intake)
       end
 
       def calculate_line_40a
         case @intake.household_rent_own
         when "own"
-          if @intake.property_tax_paid.nil?
+          if @intake.property_tax_paid.nil? || @intake.property_tax_paid.zero?
             return nil
           end
           property_tax_paid = @intake.property_tax_paid
         when "rent"
-          if @intake.rent_paid.nil?
+          if @intake.rent_paid.nil? || @intake.rent_paid.zero?
             return nil
```

Comment:
[dust] the above two could be reduced to

`return nil unless @intake.property_tax_paid.positive?`

and

`return nil unless @intake.rent_paid.positive?`

Reply:
Yes, I like this! Much more legible.

Reply:
Unfortunately when the value is nil, then positive? can't be called on it, so I'm leaving this as is.

Reply:
oh right. could be `unless @intake.property_tax_paid&.positive?`  then?